util/encoding/encoding_test.go

```diff
@@ -60,3 +61,38 @@ func TestScalarHexString(t *testing.T) {
 	ErrFatal(err)
 	require.True(t, sc.Equal(s2))
 }
+
+// Tests for error cases
+type MockFailingReader struct {
+	data []byte
+}
+type MockEmptyReader struct {
+	data []byte
+}
+
+func (m *MockFailingReader) Read(p []byte) (n int, err error) {
+	return copy(p, m.data), io.EOF
+}
+func (m *MockEmptyReader) Read(p []byte) (n int, err error) {
+	return 0, nil
+}
+
+func TestReadHexPointError(t *testing.T) {
+	// Test case: invalid hex encoding
+	reader := bytes.NewReader([]byte("invalidhex"))
+	_, err := ReadHexPoint(s, reader)
+	require.Error(t, err, "Expected error when reading invalid hex encoding, but got nil")
+
+	// Test case: reader fails
+	mockReader1 := &MockFailingReader{data: []byte("abc")}
+	_, err = ReadHexPoint(s, mockReader1)
+	require.Error(t, err, "Expected error when reader fails, but got nil")
+
+	// Test case: not enough bytes from stream
+	mockReader2 := &MockEmptyReader{data: []byte("abc")}
+	_, err = ReadHexPoint(s, mockReader2)
+	require.Error(t, err, "Expected error when not enough bytes from stream, but got nil")
+	if err.Error() != "didn't get enough bytes from stream" {
+		t.Errorf("Expected error message: didn't get enough bytes from stream, but got %s", err.Error())
```

Comment:
This could be done using `require.ErrorContains`.

Reply:
I used `require.EqualError` to exactly match it